bionic/flow.py

```diff
@@ -843,11 +854,40 @@ def entity_protocol(self, name):
         ----------
 
         name: String
-            The name of a entity.
+            The name of an entity.
         """
 
         return self._state.get_provider(name).protocol_for_name(name)
 
+    def entity_docstring(self, name):
+        """
+        Returns the docstring for entity ``name`` if a docstring is defined,
+        otherwise return None.
+
+        Parameters
+        ----------
+
+        name: String
+            The name of an entity.
+        """
+        return self._state.get_provider(name).docstring
+
+    def set_entity_docstring(self, name, docstring):
```

Comment:
In general we don't have setters for entity properties like this.  If we did want to provide this, we'd need to make sure to do it in a way that doesn't mutate any existing flow objects, which are supposed to be immutable.  E.g.,

```
flow = builder.build()
# This would also modify the flow we just created, right?
builder.set_entity_docstring('entity', 'new docstring')
```

However, as far as I can tell there's not much reason to let people mutate properties on existing entities.  (Disagree?)  That's why we only allow flow mutation in fairly constrained ways -- and also why a lot of our internal objects are immutable.

Reply:
Yeah, this makes total sense, especially in light of your comments below about `ShortcutProxy`!